### DISCUSSION TITLE
Estimated shipping dates in the Cart/Checkout block

### DISCUSSION THREAD

**Discussion by opr** [woocommerce/woocommerce#45024] [WooCommerce, Blocks, and Site Editing]
This discussion is for the use-case of shipping methods that may want to show an estimated delivery time alongside their rates. Here is an example of what this may look like.

![image](https://github.com/woocommerce/woocommerce-blocks/assets/5656702/00d5bbcb-4dad-487c-88ac-dfff0ac6226a)

We will add information to this discussion as we get more details on how we plan to implement this extensibility point.

**Requirements**

- Shipping extensions must be able to specify an estimated delivery date, or range of dates
- This estimate must be shown in the shipping methods block. It should show regardless of whether the rate is selected or not.
- It should not be too distracting from the rest of the checkout process (keep it text only maybe)
- Shipping methods should not be able to interfere with rates that do not belong to that method.

**Comment by BrianHenryIE** [comment]
The estimated delivery date is not necessarily determined by the shipping plugin.

E.g. I have a plugin [bh-wc-shipment-tracking-updates](https://github.com/brianHenryIE/bh-wc-shipment-tracking-updates) which does not provide a shipping method whatsoever, but does contain all the data necessary to calculate a shipping estimate.

**Reply by opr** [reply]
Thanks for the additional use case! I'll rethink this approach and update the post. One of the requirements I came up with is:

> Shipping methods should not be able to interfere with rates that do not belong to that method.

I might have been being overly cautious here and it might be OK to drop this since I think we're going to want to keep it text-only.
